packages/discord.js/src/client/BaseClient.js

```diff
@@ -14,6 +14,12 @@ class BaseClient extends EventEmitter {
   constructor(options = {}) {
     super({ captureRejections: true });
 
+    if (options.rest?.userAgentAppendix !== undefined) {
+      // Merging the default options when a custom user agent appendix is supplied
+      // Replaces the discord.js string. Enforce it.
+      options.rest.userAgentAppendix = `${Options.userAgentAppendix} ${options.rest?.userAgentAppendix}`;
```

Comment:
The existence of the field is already checked by line 17.
```suggestion
      options.rest.userAgentAppendix = `${Options.userAgentAppendix} ${options.rest.userAgentAppendix}`;
```